internal/appsec/waf.go

```diff
@@ -82,6 +82,12 @@ func registerWAF(rules []byte, timeout time.Duration, limiter Limiter) (unreg dy
 // newWAFEventListener returns the WAF event listener to register in order to enable it.
 func newHTTPWAFEventListener(handle *waf.Handle, addresses []string, timeout time.Duration, limiter Limiter) dyngo.EventListener {
 	return httpsec.OnHandlerOperationStart(func(op *httpsec.Operation, args httpsec.HandlerOperationArgs) {
+		var body interface{}
+		op.On(httpsec.OnSDKBodyOperationStart(func(op *httpsec.SDKBodyOperation, args httpsec.SDKBodyOperationArgs) {
+			op.On(httpsec.OnSDKBodyOperationFinish(func(operation *httpsec.SDKBodyOperation, res httpsec.SDKBodyOperationRes) {
+				body = args.Body
+			}))
```

Comment:
You don't need to listen for the finish operation here:
```suggestion
			body = args.Body
```